internal/lsp/server.go

```diff
@@ -800,6 +803,56 @@ func (l *LanguageServer) handleTextDocumentDidChange(
 	return struct{}{}, nil
 }
 
+func (l *LanguageServer) handleTextDocumentDidSave(
+	ctx context.Context,
+	_ *jsonrpc2.Conn,
+	req *jsonrpc2.Request,
+) (result any, err error) {
+	var params types.TextDocumentDidSaveParams
+	if err := json.Unmarshal(*req.Params, &params); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal params: %w", err)
+	}
+
+	if params.Text != nil && l.loadedConfig != nil {
+		if !strings.Contains(*params.Text, "\r\n") {
+			return struct{}{}, nil
+		}
+
+		enabled, err := linter.NewLinter().WithUserConfig(*l.loadedConfig).DetermineEnabledRules(ctx)
+		if err != nil {
+			l.logError(fmt.Errorf("failed to determine enabled rules: %w", err))
+
+			return struct{}{}, nil
+		}
+
+		formattingEnabled := false
+
+		for _, rule := range enabled {
+			if rule == "opa-fmt" || rule == "use-rego-v1" {
+				formattingEnabled = true
+
+				break
+			}
+		}
+
+		resp := types.ShowMessageParams{
```

Comment:
I guess we only need to make this resp if formatting is on?

Reply:
heh, yeah, well spotted. I'll have it moved 👍